### ISSUE TITLE
Camera not working on OP3/Nougat

### ISSUE THREAD

**Issue by 1n9i9c7om** [offensive-security/kali-nethunter#843]
### Device: 
OnePlus 3

### OS version (KitKat/Lollipop/Marshmallow/Nougat): 
Nougat 7.1.1, Open Beta 13

### Built from repo (date and build command) or downloaded from website (links):
https://build.nethunter.com/nightly/3.15.4-20170211-0304/kernel-nethunter-oneplus3-nougat-3.15.4-20170211-0304.zip
https://build.nethunter.com/nightly/3.15.4-20170211-0304/nethunter-generic-arm64-kalifs-full-rolling-3.15.4-20170211-0304.zip

### Output of `cat /proc/version` in adb shell or NetHunter Terminal:
Linux version 3.18.47-nethunter-3.1 (jc@westerlay) (gcc version 6.2.1 20161016 (Linaro GCC 6.2-2016.11) ) #1 SMP PREEMPT Thu Jan 26 22:15:43 EST 2017

## Issue: 
Camera won't work at all. The official, pre-installed camera app crashes instantly and shows "Unknown error" in a toast notification,
Open Camera won't crash, but says that there's been an error "opening the camera".

**Comment by 1n9i9c7om** [comment]
The build I'm using [should be the one that got a camera fix](https://github.com/offensive-security/kali-nethunter/issues/765#issuecomment-275773804), right?

If so, I feel like the new camera update that came with [Open Beta 13](https://forums.oneplus.net/threads/oxygenos-open-beta-13-for-oneplus-3.511083/) might conflict with something.

**Comment by ThoughtfulDev** [comment]
@1n9i9c7om have the same Problem with OOS 4.1.3 any news how to fix this?

**Comment by xvovx** [comment]
Having the same issues with identical everything from @1n9i9c7om OP, but on my 3t.

Has anyone sorted this yet?
@jcadduono said it was fixed on Jan 27th but...

**Comment by Simbaclaws** [comment]
Having the exact same issues on my oneplus 3T. Identical to OP aswell.

**Comment by shonenx333** [comment]
@1n9i9c7om @xvovx @Simbaclaws i just used the unofficial boeffla builds (https://forum.xda-developers.com/oneplus-3/development/kernel-boeffla-kernel-linaro-uber-t3455888) works just as nice (apart from getting the wlan1 usb dongle to work) everything works (yes also the camera :D).

But for the usb wifi dongle with this kernel a solution with the boeffla kernel is mentoined here (https://forum.xda-developers.com/oneplus-3/development/kernel-boeffla-kernel-linaro-uber-t3455888/page31) but for me some modules are missing.

TL;DR: Use this kernel https://forum.xda-developers.com/oneplus-3/development/kernel-boeffla-kernel-linaro-uber-t3455888 (everythign works except usb wifi)

**Comment by Simbaclaws** [comment]
@shonenx333 did you flash that kernel over the existing nethunter kernel with something like twrp?
I'm hoping this won't cause any breakage with the kali chroot or nethunter app... If it doesn't and I can easilly flash it over the existing nethunter kernel that would be awesome.

EDIT: I'm also noticing other strange functionality with the nethunter kernel. For example some apps don't seem to have access to my sdcard. For example screenshots don't show up at all in whatsappp and I can't send them to other people. Neither can I access my sdcard in solid explorer. I'm noticing very strange behaviour so far. Hopefully the boeffla kernel will fix the issue?

**Comment by ThoughtfulDev** [comment]
@Simbaclaws yeah kind of I used the update nethunter mentioned on page 31 and then flashed the kernel. But as always take a Nandroid backup please. 

I haven't seen any issues regarding the sdcard on nethunter or boeffla kernel.. Sorry

**Comment by Simbaclaws** [comment]
@ThoughtfulDev
Thanks for the help, the boeffla kernel seems to have solved all of my problems :+1:

**Comment by chhc007** [comment]
@Simbaclaws
Hey,I have a op3t too
I really want to know which kernel you are using ? Does the camera ,usb wifi dongle work?
If you can use them I really hope you can tell me how the detail.

**Comment by websafe** [comment]
Same problem here, 3T, Build: ONEPLUS A3003_28_171111

**Comment by randoomguy** [comment]
Hi all !
Have anyone ever sorted this issue ?
Have the same on OP5, after oreo update camera, flashlight, youtube not working.
tried downgrade, custom roms, resets etc but no joy :/

Thanks
